dotCMS/src/main/java/com/dotcms/storage/FileSystemStoragePersistenceAPIImpl.java

```diff
@@ -0,0 +1,346 @@
+                        rootFolder));
+    }
+
+
+
+    /**
+     * Adds a mapping between a bucket name and a file
+     *
+     * @param groupName {@link String} bucket name
+     * @param folder {@link File}
+     */
+    void addGroupMapping(final String groupName, final File folder) {
+        if (!folder.isDirectory() || !folder.exists() || !folder.canWrite()) {
+            throw new IllegalArgumentException(String.format(
+                    "Invalid attempt of mapping an non existing or writable folder. Argument`%s` must be a valid. ",
+                    folder));
+        }
+        this.groups.put(groupName, folder);
+        Logger.info(FileSystemStoragePersistenceAPIImpl.class, () -> String.format("Registering New Group with key is `%s` mapped to folder `%s` ",groupName, folder));
+    }
+
+    @Override
+    public boolean existsGroup(final String groupName) {
+        return this.groups.containsKey(groupName) && this.groups.get(groupName).exists();
+    }
+
+    @Override
+    public boolean existsObject(final String groupName, final String objectPath) {
+
+        return this.existsGroup(groupName) && new File(this.groups.get(groupName), objectPath)
+                .exists();
+    }
+
+    @Override
+    public boolean createGroup(final String groupName) {
+        return this.createGroup(groupName, ImmutableMap.of());
+    }
+
+    @Override
+    public boolean createGroup(final String groupName, final Map<String, Object> extraOptions) {
+        final File rootGroup = this.groups.get(getRootGroupKey());
+        final File destBucketFile = new File(rootGroup, groupName);
+        final boolean mkdirs = destBucketFile.mkdirs();
+        if(mkdirs) {
+           this.groups.put(groupName, destBucketFile);
+        }
+        return mkdirs;
+    }
+
+    @Override
+    public int deleteGroup(final String groupName) {
+        final File rootGroup = this.groups.get(getRootGroupKey());
+        final File destBucketFile = new File(rootGroup, groupName);
+        if (!rootGroup.equals(destBucketFile)) {
+            final int count = countFiles(destBucketFile);
+            FileUtil.deltree(destBucketFile, true);
+            return count;
+        }
+        return 0;
+    }
+
+    public boolean deleteObject(final String groupName, final String path) {
+        return new File(this.groups.get(groupName), path).delete();
+    }
+
+    @Override
+    public List<String> listGroups() {
+
+        return new ImmutableList.Builder<String>().addAll(this.groups.keySet()).build();
+    }
+
+    @Override
+    public Object pushFile(final String groupName,
+            final String path,
+            final File file,
+            final Map<String, Object> extraMeta) {
+
+        if (!this.existsGroup(groupName)) {
+
+            throw new IllegalArgumentException("The groupName: " + groupName +
+                    ", does not have any file mapped");
```

Comment:
Codacy found an issue: [The String literal ", does not have any file mapped" appears 4 times in this file; the first occurrence is on line 126](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6220272)